src/main/java/lotto/model/LottoNumberValidator.java

```diff
@@ -0,0 +1,42 @@
+package lotto.model;
+
+import static lotto.constant.NumberConstant.LOTTO_MAX_NUMBER;
+import static lotto.constant.NumberConstant.LOTTO_MIN_NUMBER;
+import static lotto.constant.NumberConstant.LOTTO_SIZE;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lotto.enums.ErrorMessage;
+
+public class LottoNumberValidator {
+
+    public static void validateNumbersSize(List<Integer> numbers) throws IllegalArgumentException {
+        if (numbers.size() != LOTTO_SIZE) {
+            throw new IllegalArgumentException(
+                ErrorMessage.INVALID_NUMBERS_COUNT_ERROR.getMessage());
+        }
+    }
+
+    public static void validateNumberRange(List<Integer> numbers) throws IllegalArgumentException {
+        boolean isOutOfRange = numbers.stream().anyMatch(number -> number < LOTTO_MIN_NUMBER
+            || number > LOTTO_MAX_NUMBER);
```

Comment:
조건문이 논리 연산자로 결합되는 순간 가독성이 떨어진다고 하더라고요.
이 경우, 해당 조건 자체를 `Predicate`, 또는 `boolean`을 반환하는 메서드로 추출하면 가독성이 훨씬 좋아집니다!
아래는 해당 클래스에 `private boolean isNumberInRange(int number)` 라는 메서드가 존재했다는 가정 하의 리팩토링입니다.
```suggestion
        boolean isOutOfRange = numbers.stream()
                .anyMatch(this::isNumberInRange);
```